### DISCUSSION TITLE
JWT signature: When to use RSA, EC or secret?

### DISCUSSION THREAD

**Discussion by alejandropg** [micronaut-projects/micronaut-security#1317] [Q&A]
Hi,

I would like to know if there is any recommendation about when to use JWT signed with RSA, EC, or a secret.

I mean is there any guide about it, or any recommendation about the use case, for example, concerning the security or the performance...  I would like to know the better implementation to use in each case.

I searched in the Micronaut documentation and some googling, but I didn't get anything clear.

Thanks a lot!

**Comment by sdelamo** [comment]
Micronaut Security JWT uses under the hood Nimbus JOSE. They published a [selection guide](https://connect2id.com/products/nimbus-jose-jwt/algorithm-selection-guide)
